pkg/asset/installconfig/aws/instancetypes.go

```diff
@@ -0,0 +1,60 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	// "github.com/davecgh/go-spew/spew"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+)
+
+const (
+	// Minimum resource requirements
+	minimumControlPlaneVCpus  int64 = 4
+	minimumControlPlaneMemory int64 = 16384
+	minimumComputeVCpus       int64 = 2
+	minimumComputeMemory      int64 = 8192
+)
+
+// validateInstanceType validates a provided instance type meets minimum resource requirements.
+// It returns any errors from validation.
+func validateInstanceType(ctx context.Context, sess *session.Session, region string, InstanceType string, fldPath *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	// spew.Dump(sess)
+	client := ec2.New(sess, aws.NewConfig().WithRegion(region))
+	resp, err := client.DescribeInstanceTypesWithContext(ctx, &ec2.DescribeInstanceTypesInput{InstanceTypes: []*string{&InstanceType}})
+	if err != nil {
+		return append(allErrs, field.InternalError(fldPath, err))
+	}
+
+	// need to check if this is for compute or controlplane
+	poolType := fldPath.Root().String()
+	switch poolType {
+	case "compute":
+		if *resp.InstanceTypes[0].VCpuInfo.DefaultVCpus < minimumComputeVCpus {
+			errMsg := fmt.Sprintf("Instance type %s does not meet minimum %s resource requirements of %d vCPUs", InstanceType, poolType, minimumComputeVCpus)
```

Comment:
In Go I think we prefer the error messages to not start with capital letters as it doesn't read well when wrapped.

Reply:
Updated.